src-ui/lib/components/BooksTable.svelte

```diff
@@ -42,7 +50,9 @@
         ? target.innerText.split(',').map((author) => author.trim())
         : target.innerText.trim()
 
-    books.edit({ ...book, [field]: value })
+    return booksStorePromise.then(async (booksStore) => {
```

Comment:
Note: we aren't actually _waiting_ on the promise to resolve and then do the edit, we're just triggering the effect and believing it will eventually resolve.

I'm not a huge fan of how this pattern works and looks like -- but it's partly this way because we potentially don't have a guaranteed `booksStore` before we get to this code (technically that shouldn't happen -- but you can imagine it happening on an `add` action, for example).